src/rpc/misc.cpp

```diff
@@ -251,14 +258,30 @@ static RPCHelpMan deriveaddresses()
                 }
             },
             RPCExamples{
-                "First three native segwit receive addresses\n" +
+                "First three native segwit receive addresses:\n" +
                 HelpExampleCli("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\" \"[0,2]\"") +
-                HelpExampleRpc("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\", \"[0,2]\"")
+                HelpExampleRpc("deriveaddresses", "\"" + EXAMPLE_DESCRIPTOR + "\", \"[0,2]\"") +
+                "Derive all addresses from a WIF, which has a built-in checksum:\n" +
```

Comment:
I don't think we should have examples without a sane use case. Couldn't this example just as well derive only a single address?

Reply:
Is it insane though? A user might not remember which address type they used, which is what `combo` is for. Though I guess it's likely to have been `pkh()`

Reply:
It's literally the use case I needed to figure out the various permutations for the test case I was writing. I can agree that it's probably *unusual* but still sane.

Reply:
I thought `combo` was for backward compatibility with the undesirable behaviour of legacy wallets being unable to differentiate?

In any case, having it as an example is effectively encouraging it or suggesting it, which really doesn't make sense.

Reply:
I think a frequently asked question is "how do I get the bitcoin address(es) from a given private key", actually, so providing an answer to that question in the documentation doesn't seem off to me, even if it's not exactly best practice to do that on the regular. I personally had no idea about `combo` and ended up wasting my time writing #24160 for no good reason.

Reply:
If somebody dumped a private key from an old core wallet, rather than make a regular backup, they might be in this situation where they used the SegWit address. There might also be other wallets out there that use WIF keys without specifying the address (though I suspect most adopted BIP39 before SegWit).

That said, it might be still be better to recommend that users specify which address type they want to derive, which you could illustrate with two examples, e.g.  `pkh(WIF)` and `sh(wpkh(WIF)`.

Reply:
I'm fine with that (I think `pkh` example is sufficient though), but I don't know if that addresses @luke-jr's concerns.

Reply:
`pkh` would indeed "derive only a single address"

Reply:
Switched to `pkh()` example.